### PR TITLE
Optional loss weighting and fix for splitting

### PR DESCRIPTION
**Pull request by tmachnitzki**
- outsource stepping code
- make weighting of losses possible
- fix data splitting
- add tensorboard to env

### CODE REVIEW

**Comment by tmachnitzki** [comment]
solves #7 
solves #4

File: src/models/LightningBaseModel.py [hunk 1]
```diff
@@ -13,16 +13,38 @@
...
+            if label in label_weights_dict.keys():
+                label_weights_dict[label] += 1
+            else:
+                label_weights_dict[label] = 1
+
+        weights = []
+        for label in self.class_labels:
+            weights.append(1 / label_weights_dict[label])
+
+        weight_tensor = torch.tensor(weights, dtype=torch.float32, device=self.device)
```
Comment (dgreenberg) [inline src/models/LightningBaseModel.py:44]: perhaps normalize so the mean weight is 1?

File: train_on_strand.sh [hunk 2]
```diff
@@ -22,4 +22,4 @@
 
 module load compilers/cuda/11.0
 nvidia-smi
-srun /gpfs/home/machnitz/miniconda3/envs/plankton/bin/python main.py -f "default_config.yaml"
\ No newline at end of file
+srun /gpfs/home/machnitz/miniconda3/envs/plankton/bin/python main.py -f "strand_config.yaml"
```
Comment (dgreenberg) [inline train_on_strand.sh:25]: remove this?
